last task/bot.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"strings"
+
+	"net/http"
+	"net/url"
+
+	"time"
+
+	tgbotapi "github.com/Syfaro/telegram-bot-api"
+)
+
+const (
+	dumpRaw = false
+	api     = " "
+)
+
+func isapikeyexist(api string) error {
+	if api == " " {
+		return errors.New("invalid api key")
+	}
+	return nil
+}
+
+var city = ""
+var str = ""
+
+func telegrambot() {
+	bot, err := tgbotapi.NewBotAPI("1990027676:AAGkEFcn1FDDPtKY0iBI9zGNuyYH7tTipLs")
+	if err != nil {
+		log.Panic(err)
+	}
+
+	bot.Debug = true
+	log.Printf("Authorized on account %s", bot.Self.UserName)
+	var ucfg tgbotapi.UpdateConfig = tgbotapi.NewUpdate(0)
+	ucfg.Timeout = 60
+	upd, _ := bot.GetUpdatesChan(ucfg)
+	for {
+		select {
+		case update := <-upd:
+			ChatID := update.Message.Chat.ID
+			Text := ""
+			switch update.Message.Command() {
+			case "city":
+				Text = update.Message.Text
+				out := strings.Replace(Text, "/city", "", -1)
+				city = out
+				urlString := fmt.Sprintf("http://api.openweathermap.org/data/2.5/weather?q=%s&APPID=%s", city, api)
+				u, err := url.Parse(urlString)
+				res, err := http.Get(u.String())
+				if err != nil {
+					log.Fatal(err)
+				}
+				jsonBlob, err := ioutil.ReadAll(res.Body)
+				res.Body.Close()
+				if err != nil {
+					log.Fatal(err)
+				}
+
+				var data map[string]interface{}
+				if dumpRaw {
+					fmt.Printf("blob = %s\n\n", jsonBlob)
+				}
+				err = json.Unmarshal(jsonBlob, &data)
+				if err != nil {
+					fmt.Println("error:", err)
+				}
+				data = deletuselessinf(data)
```

Comment:
Why you just don't operate with golang structs without doing some hacks?=)
Try https://github.com/jmoiron/jsonq for JSON access=)